group_vars/betabarrel_cluster/ip_addresses.yml

```diff
@@ -1,10 +1,22 @@
 ---
 ip_addresses:
+  bb-porch:
+    bb_internal_management:
+      address: 10.10.1.164
+      netmask: /32
+      publicly_exposed: true  # This internal IP is linked to a public (floating) IP.
+    vlan16:
+      address: 195.169.22.140
+      netmask: /32
+      fqdn: 'NXDOMAIN'
   betabarrel:
     vlan983:
       address: 172.23.41.225
-      netmask: /23
```

Comment:
This is a bit confusing. The vlan983 network segment uses /23, but here we specify a single IP address. We use the same `ip_addresses` data structure in several places to make it easier to parse it with the same code/logic. Therefore the `ip_addresses` variable can contain both info for networks as well as info for individual IP addresses for a specific machine. All of that can then be parsed with the same code for configuring `iptables` firewalls or OpenStack _security groups_. In this case the var contains a single IP for a NIC of the `betabarrel` machine: that must be /32 to make it a single IP. In other words the value for `netmask` listed here is not used to configure the netmask for the NIC on `betabarrel`. If you want to configure iptables of an OpenStack security group to allow access to/from the whole VLAN it should be 172.23.41.0/23, but if you want to allow access only to/from `betabarrel` it should be 172.23.41.225/32.

Reply:
Well that is quit confusing, yes.
If we talk about the BB CentOS interface itself: network range was previously defined at `/24`, which prevented BB to access the network. That has nothing to do with the network configuration itself.

> In other words the value for netmask listed here is not used to configure the netmask for the NIC on betabarrel.
This is particularly confusing, I think that then it should either:
- not be defined there, but somewhere else,
- be defined with another variable name
- or not being defined at all, and be by default `/32` - it that's expected.

I think that it would be less confusin in general if
1. in `static_inventories/...yml` machine's configuration sets the `netmask` on the machine's interface only - on CentOS level
2. if machine has in static inventories also defined `netmask`, it should be used inside CentOS configuration for the appropriate interface as network range
3. `netmask` for openstack networking should be configured in the network configuration (inside `vars.yml`) and used by openstack (and later if needed in firewal)